ledger/participant-integration-api/src/main/scala/platform/indexer/package.scala

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform
+
+import com.daml.ledger.resources.ResourceOwner
+
+import scala.concurrent.Future
+
+package object indexer {
+
+  /** Indexer is a factory for indexing. Future[Unit] is the completion Future, as it completes indexing is completed with results accordingly (Success/Failure)
```

Comment:
:clap: thanks for that, that was unintentional. I was not paying enough attention on mutable, and since initialization in mutable is on the same stage as resetting the schema, I thought they must belong outside of recoveringindexer.
Looks like I was wrong, and we reset every time when we recover :thinking: which must be a bug from far before, got fooled by it.
Funny how it could go unnoticed so far :) probably because not much recovering is needed in sandboxnext ;)
It sort of makes sense that still all tests pass: we do not really have high level integration tests triggering recovery...would be nice to have such

Reply:
> This line left me with many questions. I think it needs a better documentation. What are the expectations of Indexer? Does Indexer.acquire initialize the database (i.e., write the ledger id, and/or clean up partially ingested updates)? Can the same Indexer start indexing more than once? If yes, what offset does it restart the indexing from?

I hope you don't suggest to add all this to the description of the Indexer alias :)
Happy to take suggestions!

Reply:
> we reset every time when we recover 

Do we? The way I read it:
- `StandaloneIndexerServer.startIndexer()` runs only once
- `indexerFactory.initialized(resetSchema).acquire()` runs only once
- `JdbcIndexer.Factory.initializedMutatingSchema()` runs only once
  - The database is reset only once, and the start offset is queried only once
- `JdbcIndexer.acquire()` runs any time the `RecoveringIndexer` recovers , but each time it starts streaming state updates from the same offset

Reply:
> I hope you don't suggest to add all this to the description of the Indexer alias :)

Actually, I was. Maybe not these exact words, but I feel a detailed description helps with understanding the chains of `ResourceOwner`/`Resource`/`Future` we use. It took me a long time to understand what the `Indexer` trait really is, and I'm still not sure I got it right.

Reply:
> Do we? The way I read it: ...
Your analyses seems correct to me for the changed code. I was referring to the code before this PR. AFAICS that was doing reset on every recovery-indexer restart (as initialization as well) -- this prrevious situation which made me think it is okay to initialize only at the beginning, and not on every recovery-indexer restart

Reply:
> Actually, I was. Maybe not these exact words, but I feel a detailed description helps with understanding the chains of ResourceOwner/Resource/Future we use. It took me a long time to understand what the Indexer trait really is, and I'm still not sure I got it right.

I see
* I understand you confusion. It took me quite some time to understand the original Indexer, also made thinks very hard to understand because of it. This PRs one main objective is to make a step to simplify that.
* For my part I would be okay to keep the documentation simple right now, as I think about this as a step towards cleanup indexer factories.
* If you think even in this PR it would be beneficial to have more elaborate documentation on the Indexer alias, I am happy to improve it

Reply:
Misunderstandings solved offline, the one uncovered issue was fixed in this PR.